pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -346,12 +348,13 @@ func (s *ServiceEntryStore) Services() ([]*model.Service, error) {
 		services = append(services, convertServices(cfg)...)
 	}
 
-	return services, nil
+	return autoAllocateIPs(services), nil
```

Comment:
before merge - probably need a feature flag for this.

Does this have any impact if we don't have DNS capture?

Reply:
Nope. Thats why I didn't put a feature flag. All the auto allocated IPs are in the AutoAllocatedAddress field which is not used by GetServiceAddressForProxy unless the DNS_CAPTURE meta is set.